### PR TITLE
[AOTI] Use `len(serialized_weights)` when calculating `consts_size`

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139054

Fixes the failure of INT8 DLRM using AOTI.
The previous code calculates `consts_size` directly using `tensor` from `graph.constants`:
```
  consts_size = sum(
      get_nbytes_of_tensor(tensor, all_cuda)
      for (name, tensor) in graph.constants.items()
      if name not in graph.folded_constants
  )
```
Meanwhile, the actual bytes to serialize (`serialized_weights`) is using `graph.get_original_value_of_constant(name)`:
```
  serialized_weights = b"".join(
      _to_bytes(graph.get_original_value_of_constant(name), all_cuda)
      for name in graph.constants.keys()
      if name not in graph.folded_constants
  )
```

`tensor` from `graph.constants` could be different from `graph.get_original_value_of_constant(name)` thus making the `consts_size` inconsistent with the actual byte size of the `serialized_weights`, resulting in runtime error `weights_offset must be aligned to 16K boundary`, similar to what happened in https://github.com/pytorch/pytorch/pull/135205.

This PR direclty gets `consts_size ` using `len(serialized_weights)`, which fixes the inconsistency.

We also added a `reduce_range` argument to the `get_default_x86_inductor_quantization_config` function, which is needed in the unit test to avoid accuracy issue on CI machines (earlier CPUs without VNNI).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: test/inductor/test_aot_inductor.py [hunk 1]
```diff
@@ -3882,6 +3917,7 @@ def fail_cuda(is_skip=False):
 
 class AOTInductorTestABICompatibleCpu(AOTITestCase):
     device = "cpu"
+    device_type = "cpu"
```
Comment (chunyuan-w) [inline test/inductor/test_aot_inductor.py:3920]: `device_type` is needed in `skipCUDAIf`

File: test/inductor/test_aot_inductor.py [hunk 2]
```diff
@@ -3900,6 +3936,7 @@ class AOTInductorTestABICompatibleCpu(AOTITestCase):
 @unittest.skipIf(sys.platform == "darwin", "No CUDA on MacOS")
 class AOTInductorTestABICompatibleCuda(AOTITestCase):
     device = "cuda"
+    device_type = "cuda"
```
Comment (chunyuan-w) [inline test/inductor/test_aot_inductor.py:3939]: ditto

File: test/inductor/test_aot_inductor_arrayref.py [hunk 3]
```diff
@@ -197,6 +198,7 @@ class AOTInductorTestABICompatibleCpuWithStackAllocationAndMinimalArrayRefInterf
     TestCase
 ):
     device = "cpu"
+    device_type = "cpu"
```
Comment (chunyuan-w) [inline test/inductor/test_aot_inductor_arrayref.py:201]: ditto

**Review by leslie-fang-intel — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Comment by yushangdi** [comment]
@chunyuan-w Hi, I noticed that this PR uses `capture_pre_autograd_graph`. Note that this API is deprecated and we plan to delete it very soon. Please use `torch.export.export_for_training(...).module()` instead moving forward. Thanks!
